src/crypto.ts

```diff
@@ -195,4 +200,57 @@ export default class Crypto {
       decodeBase64(formSecretKey)
     )
   }
+
+  /**
+   * Decrypts an encrypted submission, and also download and decrypt any attachments alongside it.
+   * @param formSecretKey Secret key as a base-64 string
+   * @param decryptParams The params containing encrypted content and information.
+   * @returns A promise of the decrypted submission, including attachments (if any). Or else returns null if a decryption error decrypting any part of the submission.
+   * @throws {MissingPublicKeyError} if a public key is not provided when instantiating this class and is needed for verifying signed content.
+   */
+  decryptWithAttachments = async (
+    formSecretKey: string,
+    decryptParams: DecryptParams
+  ): Promise<DecryptedContentAndAttachments | null> => {
+    const decryptedRecords: DecryptedAttachments = {}
+    const filenames: Record<string, string> = {}
+
+    const attachmentRecords: EncryptedAttachmentRecords = decryptParams.attachmentDownloadUrls ?? {}
+    const decryptedContent = this.decrypt(formSecretKey, decryptParams)
+    if (decryptedContent === null) return null
+
+    // Retrieve all original filenames for attachments for easy lookup
+    for (const i in decryptedContent.responses) {
+      const response = decryptedContent.responses[i]
+      if (response.fieldType === 'attachment' && response.answer) {
+        filenames[response._id] = response.answer
+      }
+    }
+
+    for (let fieldId in attachmentRecords) {
+      const downloadResponse = await axios.get(attachmentRecords[fieldId], { responseType: 'json' })
+      if (downloadResponse.status !== 200) return null
```

Comment:
i think how this works is that the for loop blocks on each iteration until the network call has completed. This might be significant in the future, especially because network calls are relatively expensive.

would it be better to instead `map` this to an `axios.get` request and then use `Promise.all` /`Promise.allSettled` so that requests can be executed in parallel?

Reply:
`Promise.all` would be good, the function should fail when a single attachment cannot be fetched